recover.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+	"runtime"
+)
+
+// A recoverMux recovers from panics during an http.Handler, sending an error
+// response to a client in the event a panic occurs.
+type recoverMux struct {
+	mux *http.ServeMux
```

Comment:
`nit` might consider embedding the servemux to indicate the `is-a` relationship

Reply:
After looking at https://golang.org/pkg/net/http/#ServeMux, it looks like the standard library wants to initialize this itself and simply give you a pointer to a struct with unexported fields. It doesn't look like initializing and embedding it will work without copying it which seems bad.

Reply:
:+1: